src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
```

Comment:
Same as above, if-else syntax would be more readable, reduce could be used, division by 0 should be handled.
```js
function calculateAverageAgeDiff(people, onlyWithSon) {
  let children = [];

  if (onlyWithSon) {
    children = people.filter(p => p.sex === 'm' && people.some(m => m.name === p.mother));
  } else {
    children = people.filter(p => people.some(m => m.name === p.mother));
  }

  if (children.length === 0) {
    return 0; // Handle division by zero
  }

  const sum = children.reduce((acc, ch) => {
    const mother = people.find((m) => m.name === ch.mother);
    return acc + (ch.born - mother.born);
  }, 0);

  return sum / children.length;
}
```